adapters/repos/db/vector/hnsw/distancer/cosine_dist.go

```diff
@@ -53,3 +53,11 @@ func (d CosineDistanceProvider) Type() string {
 func (d CosineDistanceProvider) New(a []float32) Distancer {
 	return &CosineDistance{a: a}
 }
+
+func (d CosineDistanceProvider) Step(x, y float32) float32 {
```

Comment:
I think we can git rid of Step() function since it is equiavalent to x*y.
I think the dot product is not effient as x and y are not vector